src/main/java/com/microsoft/graph/authentication/TokenCredentialAuthProvider.java

```diff
@@ -0,0 +1,87 @@
+package com.microsoft.graph.authentication;
+
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import com.microsoft.graph.exceptions.AuthenticationException;
+import com.microsoft.graph.exceptions.Error;
+import com.microsoft.graph.httpcore.IHttpRequest;
+import okhttp3.Request;
+import com.microsoft.graph.exceptions.ErrorConstants.*;
+
+import java.util.List;
+
+public class TokenCredentialAuthProvider implements ICoreAuthenticationProvider , IAuthenticationProvider {
+
+    //TokenCredential expected form user
+    private TokenCredential tokenCredential;
+    //Context options which can be optionally set by the user
+    private TokenRequestContext context;
+    //Access token to be retrieved
+    private AccessToken accessToken;
+
+    /**
+     * Creates an Authentication provider using a passed in TokenCredential
+     *
+     * @param tokenCredential Credential object inheriting the TokenCredential interface used to instantiate the Auth Provider
+     * @throws AuthenticationException exception occurs if the TokenCredential parameter is null
+     */
+    public  TokenCredentialAuthProvider(TokenCredential tokenCredential) throws AuthenticationException {
+        if(tokenCredential == null) {
+            throw new AuthenticationException(new Error(Codes.InvalidArgument,
+                    String.format(Messages.NullParameter, "TokenCredential"))
+                    ,new IllegalArgumentException());
+        }
+
+        this.tokenCredential = tokenCredential;
+        this.context = new TokenRequestContext();
+    }
+
+    /**
+     *Created an Authentication provider using a TokenCredential and list of scopes
+     *
+     * @param tokenCredential Credential object inheriting the TokenCredential interface used to instantiate the Auth Provider
+     * @param scopes Specified desired scopes of the Auth Provider
+     * @throws AuthenticationException exception occurs if the TokenCredential parameter is null
+     */
+    public TokenCredentialAuthProvider(TokenCredential tokenCredential, List<String> scopes) throws AuthenticationException {
+        this(tokenCredential);
+        this.context.setScopes(scopes);
```

Comment:
how would I use these scopes?

Reply:
The scopes list is used to instantiate a TokenRequestContext object, which is then used by the getToken method in the credential class. TokenRequestContext are details added to the request of the getToken method when retrieving the of the AccessToken.

https://www.javadoc.io/static/com.azure/azure-core/1.7.0-beta.2/com/azure/core/credential/TokenRequestContext.html